docs/guides/http/json-validation.md

```diff
@@ -0,0 +1,93 @@
+# JSON validation
+
+Bocadillo has built-in support for JSON data validation using [TypeSystem]. It allows you to perform schema-based validation, as well as use [orm] to get RESTful validation _à la_ Django REST Framework.
+
+[typesystem]: https://www.encode.io/typesystem/
+[orm]: https://github.com/encode/orm
+
+## How it works
+
+JSON data validation is implemented by registering an [error handler](./error-handling.md) for `typesystem.ValidationError`.
+
+Thanks to this, Bocadillo will format and send a `400 Bad Request` error response if validation fails within an HTTP view — see examples below!
+
+::: tip
+TypeSystem comes installed with Bocadillo, and this feature is enabled by default, so you can use it right out of the box. 🎉
+:::
+
+## Examples
+
+### TypeSystem schema
+
+Consider the following example application: we define a `Todo` schema, and use it in an endpoint to validate the JSON payload before creating todo items:
+
+<<<@/docs/guides/snippets/json_validation/app.py
+
+Let's try it out:
+
+1. Run the app using `uvicorn example.app:app`.
+2. Make a request with a valid JSON payload — everything should be fine:
+
+```bash
+$ curl \
+  -X POST \
+  -d '{"title": "Make breakfast"}' \
+  http://localhost:8000/todos
+```
+
+```json
+{
+  "title": "Make breakfast",
+  "done": false
+}
+```
+
+- Send an invalid payload instead (e.g. title too long) — Bocadillo automatically sends back an explicit error response:
+
+```bash
+$ curl \
+  -X POST \
+  -d '{"title": "Buy cornflakes at the store and make breakfast"}' \
+  http://localhost:8000/todos
+```
+
+```json
+{
+  "error": "400 Bad Request",
+  "status": 400,
+  "detail": { "title": "Must have no more than 20 characters." }
+}
+```
+
+This is a fairly basic example, but you can read the [TypeSystem documentation][typesystem] to learn about more complex validation techniques.
+
+### `orm` model
+
+You can use JSON validation error handling with [orm] models: they're TypeSystem schemas themselves, and their `.create()` and `.update()` methods raise `ValidationError` exceptions too.
```

Comment:
Is the difference between schema-based and orm-based basically that with the schema you have to call `validate` while with orm it happens automatically when performing writes?

Reply:
Yes, the thing is that .create() and .update() make an internal call to .validate() on a schema that orm builds out of the model. As I said in the other discussion maybe this shouldn’t be mentioned here to avoid confusion?